lib/ProductOpener/Nutrition.pm

```diff
@@ -308,20 +373,1471 @@ sub convert_nutrient_to_100g ($nutrient_ref, $original_per, $original_per_quanti
+			convert_nutrition_input_sets_hash_to_array($input_sets_hash_ref, $product_ref));
+	}
+	return;
+}
+
+=head2 remove_empty_nutrient_values_and_set_unspecified_nutrients ($input_set_ref)
+
+Removes nutrients with empty values from an input set.
+
+If a nutrient has a modifier equal to "-", it means no value is specified on the packaging.
+
+The nutrient is removed from the input set and added to the unspecified_nutrients array.
+
+=head3 Arguments
+
+=head4 $input_set_ref
+
+Reference to the input set hash
+
+=cut
+
+sub remove_empty_nutrient_values_and_set_unspecified_nutrients ($input_set_ref) {
+
+	if (exists $input_set_ref->{nutrients} and ref $input_set_ref->{nutrients} eq 'HASH') {
+		foreach my $nid (sort keys %{$input_set_ref->{nutrients}}) {
+			my $nutrient_ref = $input_set_ref->{nutrients}{$nid};
+			# If we have a modifier equal to a dash - , it means no value is specified on the packaging
+			# We remove the nutrient from the input set
+			if (defined $nutrient_ref->{modifier} and $nutrient_ref->{modifier} eq '-') {
+				delete $input_set_ref->{nutrients}{$nid};
+				# add it to the unspecified_nutrients array if it's not already there
+				if (not defined $input_set_ref->{unspecified_nutrients}) {
+					$input_set_ref->{unspecified_nutrients} = [];
+				}
+				if (not grep {$_ eq $nid} @{$input_set_ref->{unspecified_nutrients}}) {
+					push @{$input_set_ref->{unspecified_nutrients}}, $nid;
+				}
+			}
+			# If the value_string is undefined or empty, we remove the nutrient from the input set
+			elsif ((not defined $nutrient_ref->{value_string}) or ($nutrient_ref->{value_string} eq '')) {
+				delete $input_set_ref->{nutrients}{$nid};
+			}
+			# If the modifier is undefined or empty, we remove it from the nutrient
+			elsif ((not defined $nutrient_ref->{modifier}) or ($nutrient_ref->{modifier} eq '')) {
+				delete $nutrient_ref->{modifier};
+			}
+		}
+	}
+	return;
+}
+
+=head2 add_nutrition_fields_from_product_to_populated_fields($product_ref, \%populated_fields, $sort_key)
+
+This function is used by Export.pm to generate the list of populated nutrition fields for a product.
+Export.pm can then create a CSV file with columns that have data for at least one product.
+
+=head3 Arguments
+
+=head4 $product_ref
+
+Reference to the product hash
+
+=head4 \%populated_fields
+
+Reference to the hash of populated fields
+
+=head4 $sort_key
+
+A string used to sort the fields in the CSV file.
+The nutrition fields sort keys will be prefixed by this sort key.
+
+=cut
+
+sub add_nutrition_fields_from_product_to_populated_fields($product_ref, $populated_fields_ref, $sort_key) {
+
+	if (not defined $product_ref->{nutrition}) {
+		return;
+	}
+
+	# Fields at the root of nutrition
+	my $item_number = 0;
+	foreach my $field ("no_nutrition_data_on_packaging") {
+		if (defined deep_get($product_ref, "nutrition", $field)) {
+			$populated_fields_ref->{$field} = $sort_key . '_0-root_' . sprintf("%02d", $item_number);
```

Comment:
In the lines below, the item_number is just after "$sort_key". Is it normal that it's different here ?